### PR TITLE
Update the ResourceData API

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
There is an undocumented semantic of Win32 Resource APIs.
The missing semantic is that all resource type/name
strings are transparently converted to uppercase
when calling any of the Win32 Resource APIs.

We don't want to apply this undocumented semantic to the reader/writer API
so we document it instead. We are avoiding applying the behavior
since ReadyToRun scenarios are designed to be a byte for byte copy
of the resource, including name as it was written by other tooling.

### CODE REVIEW

File: src/coreclr/tools/Common/Compiler/Win32Resources/ResourceData.UpdateResourceDataModel.cs [hunk 1]
```diff
@@ -90,5 +98,16 @@ private byte[] FindResourceInternal(object name, object type, ushort language)
...
             return (byte[])resLanguage.DataEntry.Clone();
         }
+
+        private static string ToUpperForResource(string str)
+        {
+            // Undocumented semantic for Win32 Resource APIs
+            StringBuilder builder = new(str.Length);
+            foreach (char c in str)
+            {
+                builder.Append('a' <= c && c <= 'z' ? char.ToUpper(c) : c);
```
Comment (jkotas) [inline src/coreclr/tools/Common/Compiler/Win32Resources/ResourceData.UpdateResourceDataModel.cs:108]: Use invariant culture?

Comment (AaronRobinsonMSFT) [reply src/coreclr/tools/Common/Compiler/Win32Resources/ResourceData.UpdateResourceDataModel.cs:108]: Not in this case. That isn't how it is defined. We can chat about this offline if you'd like.

**Review by elinor-fung — APPROVED** [review]

**Review by elinor-fung — APPROVED** [review]

**Review by MichalStrehovsky — CHANGES_REQUESTED** [review]
Isn't this introducing a new bug to the use of this in crossgen2 and ILC? Both tools have a need to _copy_ resources from the source IL assembly to the destination, without changing anything. If I'm reading this code right, we're going to uppercase whatever names are in the original and store them in the modified form. I believe it would be a observable difference with Win32 APIs like EnumResourceNames. Can this be done in a way that allows doing a straight copy?

Or... is the change to ComHost.cs in this PR all that's needed to fix the underlying bug?

We don't really need to interpret the names except for the one place that calls FindResource to find PGO data. That's our data, we fully control the casing of it. I don't think we need it to work case insensitively.

**Comment by AaronRobinsonMSFT** [comment]
> Isn't this introducing a new bug to the use of this in crossgen2 and ILC?

No. Will discuss offline.

**Review by MichalStrehovsky — APPROVED** [review]

**Comment by AaronRobinsonMSFT** [comment]
/ba-g System.Text.Json.Serialization.Tests.JsonElementTests.DeepEquals_TooDeepJsonDocument_ThrowsInsufficientExecutionStackException is unrelated.

**Pull request merged**